### ISSUE TITLE
[Feature Request] Reduce space taken up by empty options structs

### ISSUE THREAD

**Issue by seankane-msft** [Azure/azure-sdk-tools#3045]
@tg-msft "could we gray out "future proof" options/responses, render them on a single line, and not put as much extra space before/after? These cause a lot more visual noise than they're worth today."

**Comment by chlowell** [comment]
We can't gray them out, and #2998 eliminates the extraneous whitespace. We could render them on a single line though, by which I mean:
```
// today
type FooOptions struct {
    // no exported fields
}

// could do this instead
type FooOptions struct {}
```
The question then is, does explicitly calling out a struct's lack of exported fields help someone to the extent that we want to keep doing it? Or, from another angle, is `type FooOptions struct {}` intuitive or does it look like a bug? Personally, I think the single line is reasonably intuitive.

We could also make these empty structs implicit by simply not rendering their uninteresting definitions:
```
// readers can infer a ThingOptions type having no fields
type NewThing(options *ThingOptions)
```
I think that's probably too radical but it would significantly declutter some reviews.

**Comment by tg-msft** [comment]
Thanks for filing this @seankane-msft!

I agree we probably don't want to fully remove them.  But if we can't gray them out, could we do a ~~strikethrough~~ like Java does for things that have been deprecated (see the ~~`Size`~~ property at https://apiview.dev/Assemblies/Review/d1b94cf46b5a462a9f3d663794d97fc7#com.azure.storage.blob.models.Block for an example)?  Something like ~~`type FooOptions struct { /* no exported fields, here for futureproofing */ }`~~ would make it easier to skim through looking for good stuff.

**Comment by chlowell** [comment]
I'd like to save strikethrough to indicate deprecation and don't want to imply these structs are somehow dubious. They don't need to occupy so much space though, so I implemented a single line format in #2998: `type FooOptions struct {}`
